pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java

```diff
@@ -109,6 +109,9 @@
 
     protected volatile Optional<Long> topicEpoch = Optional.empty();
     private volatile boolean hasExclusiveProducer;
+    // pointer to the exclusive producer
+    private volatile Producer exclusiveProducer;
```

Comment:
Should we just use a ref to the name instead? Otherwise it will provide a much longer toString() representation

Reply:
We could.  Just wanted to preserve the existing logging behavior.

Reply:
The existing one was printing the key of the map, hence just the producer name

Reply:
changed